src/main/groovy/graph/DirectedEdge.groovy

```diff
@@ -1,6 +1,6 @@
 package graph
 
-trait DirectedEdge {
+class DirectedEdge extends Edge {
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') The class graph.DirectedEdge defines equals(Object) but not hashCode() [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=grvy%3Aorg.codenarc.rule.basic.EqualsAndHashCodeRule)